application/F3DStarter.cxx

```diff
@@ -11,7 +11,21 @@
 #include "options.h"
 #include "window.h"
 
+#include <cassert>
 #include <filesystem>
+#include <set>
+
+namespace fs = std::filesystem;
+
+namespace
+{
+long long fileModulo(long long index, long long size)
+{
+  index %= size;
+  index = index < 0 ? index + size : index;
+  return index;
```

Comment:
@Meakk, `(index + size) % size` would most probably work in the app as we're unlikely to ever decrement to lower than `-size` but technically it could still return negative values. Wouldn't it be worth it to keep the true modulo for correctness/safety?

Reply:
Well, my code has the same issue anyway.